### ISSUE TITLE
SIGSEGV on Ubuntu 18.04

### ISSUE THREAD

**Issue by Janrupf** [NovaMods/nova-renderer#126]
```
[04:37:56] [Nova Renderer/ERROR]: ERROR: API: Validation Object: VK_NULL_HANDLE (Type = 0) | vkUpdateDescriptorSets: required parameter pDescriptorWrites[0].pImageInfo[0].imageView specified as VK_NULL_HANDLE
[04:37:56] [Nova Renderer/ERROR]: ERROR: API: Validation Object: VK_NULL_HANDLE (Type = 14) | Invalid ImageView Object 0x0. The spec valid usage text states 'If descriptorType is VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE, VK_DESCRIPTOR_TYPE_STORAGE_IMAGE, or VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT, the imageView and imageLayout members of each element of pImageInfo must be a valid VkImageView and VkImageLayout, respectively' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkWriteDescriptorSet-descriptorType-00326)
[04:37:56] [Nova Renderer/ERROR]: ERROR: API: Validation Object: 0x49 (Type = 23) | vkUpdateDescriptorSets() failed write update validation for Descriptor Set 0x49 with error: Write update to descriptor in set 0x49 binding #0 failed with error message: Attempted write update to combined image sampler descriptor failed due to: Invalid VkImageView: 0. The spec valid usage text states 'If descriptorType is VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE, VK_DESCRIPTOR_TYPE_STORAGE_IMAGE, or VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT, the imageView and imageLayout members of each element of pImageInfo must be a valid VkImageView and VkImageLayout, respectively' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkWriteDescriptorSet-descriptorType-00326)
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f19208b9326, pid=24322, tid=0x00007f1997aa4700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_171-b11) (build 1.8.0_171-b11)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.171-b11 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libnvidia-glcore.so.390.48+0x13af326]
#
# Core dump written. Default location: /home/jannis/Dokumente/Coding/Minecraft/nova-renderer/jars/core or core.24322
#
# An error report file with more information is saved as:
# /home/jannis/Dokumente/Coding/Minecraft/nova-renderer/jars/hs_err_pid24322.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

When running `./gradlew runNova` this happens. Some additional information:
Graphics: GTX-770
Driver: nvidia-driver-390
Desktop: Budgie

**PS: I cloned the vulkan branch to be up to date!**

**Comment by Janrupf** [comment]
Looks like `VK_LAYER_GOOGLE_threading`, but I can't find any information how to fix this... any ideas?

**Comment by DethRaid** [comment]
Could you upload the full log? It should be in jars/loga/nova.log

**Comment by Janrupf** [comment]
And got another crash in a dev enviroment:
```
[09:48:59] [Nova Renderer/INFO]: Creating descriptor sets for pipeline Final
[09:48:59] [Nova Renderer/DEBUG]: shader_resource_manager.cpp 79 - We've created 12 sets
[09:48:59] [Nova Renderer/INFO]: Binding texture LitWorld to descriptor (set=2 binding=0)
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f3d2ca9d32a, pid=30308, tid=0x00007f3dab73e700
#
# JRE version: OpenJDK Runtime Environment (8.0_171-b11) (build 1.8.0_171-8u171-b11-0ubuntu0.18.04.1-b11)
# Java VM: OpenJDK 64-Bit Server VM (25.171-b11 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libnvidia-glcore.so.390.48+0x13af32a]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/jannis/Dokumente/Coding/Minecraft/nova-renderer/jars/hs_err_pid30308.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

**Comment by DethRaid** [comment]
Again, I need the full log

**Comment by Janrupf** [comment]
Sorry, missed your comment, here is the log: https://paste.ee/p/qPfi7

**Comment by Janrupf** [comment]
https://github.com/NovaMods/nova-renderer/blob/vulkan/src/main/cpp/render/objects/resources/shader_resource_manager.cpp#L209 is causing the segmentation fault.

**Comment by Janrupf** [comment]
Going to reopen this in another issue, figured out and fixed some weird bugs that my system caused.
